source/algorithms.tex

```diff
@@ -2904,6 +2819,148 @@
 }
 \end{codeblock}
 
+\rSec1[algorithms.results]{Algorithm result types}
+
+\pnum
+Each of the class templates specified in this subclause
+has the template parameters, data members, and special members specified below,
+and has no base classes or members other than those specified.
+
+\begin{codeblock}
+namespace std::ranges {
+  template<class I, class F>
+  struct in_fun_result {
+    [[no_unique_address]] I in;
+    [[no_unique_address]] F fun;
+
+    template<class I2, class F2>
+      requires convertible_to<const I&, I2> && convertible_to<const F&, F2>
+      constexpr operator in_fun_result<I2, F2>() const & {
+        return {in, fun};
+      }
+
+    template<class I2, class F2>
+      requires convertible_to<I, I2> && convertible_to<F, F2>
+      constexpr operator in_fun_result<I2, F2>() && {
+        return {std::move(in), std::move(fun)};
+      }
+  };
+
+  template<class I1, class I2>
+  struct in_in_result {
+    [[no_unique_address]] I1 in1;
+    [[no_unique_address]] I2 in2;
+
+    template<class II1, class II2>
+      requires convertible_to<const I1&, II1> && convertible_to<const I2&, II2>
+      constexpr operator in_in_result<II1, II2>() const & {
+        return {in1, in2};
+      }
+
+    template<class II1, class II2>
+      requires convertible_to<I1, II1> && convertible_to<I2, II2>
+      constexpr operator in_in_result<II1, II2>() && {
+        return {std::move(in1), std::move(in2)};
+      }
+  };
+
+  template<class I, class O>
+  struct in_out_result {
+    [[no_unique_address]] I in;
+    [[no_unique_address]] O out;
+
+    template<class I2, class O2>
+      requires convertible_to<const I&, I2> && convertible_to<const O&, O2>
+      constexpr operator in_out_result<I2, O2>() const & {
+        return {in, out};
+      }
+
+    template<class I2, class O2>
+      requires convertible_to<I, I2> && convertible_to<O, O2>
+      constexpr operator in_out_result<I2, O2>() && {
+        return {std::move(in), std::move(out)};
+      }
+  };
+
+  template<class I1, class I2, class O>
+  struct in_in_out_result {
+    [[no_unique_address]] I1 in1;
+    [[no_unique_address]] I2 in2;
+    [[no_unique_address]] O  out;
+
+    template<class II1, class II2, class OO>
+      requires convertible_to<const I1&, II1> &&
+        convertible_to<const I2&, II2> && convertible_to<const O&, OO>
```

Comment:
```suggestion
               convertible_to<const I2&, II2> && convertible_to<const O&, OO>
```
https://github.com/cplusplus/draft/pull/3750/files formats it like this. This occurs below too.

Reply:
FWIW, I use that style in #3750 because I'm under the impression that the editors prefer it. It is notably similar to how we indent run-on lines for template parameters.